pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,137 @@
+package testcase
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncrypt() {
+	etcdNodes, errGetEtcd := shared.GetNodesByRoles("etcd")
+	Expect(etcdNodes).NotTo(BeEmpty())
+	Expect(errGetEtcd).NotTo(HaveOccurred(), "error getting etcd nodes")
+
+	cpNodes, errGetCP := shared.GetNodesByRoles("control-plane")
+	Expect(cpNodes).NotTo(BeEmpty())
+	Expect(errGetCP).NotTo(HaveOccurred(), "error getting control plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	ips := getNodeIps(etcdNodes, cpNodes)
+
+	errSecret := shared.CreateSecret("secret1", "default")
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+
+	secretsEncryptOps("prepare", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("reencrypt", product, cpNodes[0].ExternalIP, ips)
+	secretsEncryptOps("rotate-keys", product, cpNodes[0].ExternalIP, ips)
+}
+
+func secretsEncryptOps(action, product, cpIp string, ips []string) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		time.Sleep(20 * time.Second)
+	}
+
+	for _, node := range ips {
+		var nodearr []string
+		nodearr = append(nodearr, node)
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
```

Comment:
if you are sending " nodearr" each loop iteration , why creating a slice and appending ?

Reply:
sure, but  "node" its a new var every loop right ?

why need the slice and append?

Reply:
https://go.dev/play/p/TJTUXucDSq8

Reply:
the manageservice expects []string, and node is a string.

Reply:
the manageservice was built to reboot all nodes at once. not to handle one by one.

Reply:
func shared.ManageService(product string, action string, nodeType string, ips []string) (string, error)

Reply:
got you

but still not need to append right? 


```
nodearr := []string{node}
 nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
```

Reply:
fixed.